### ISSUE TITLE
"My Jobs" dashboard is stuck forever

### ISSUE THREAD

**Issue by yarongilor** [scylladb/argus#145]
![Screenshot from 2022-07-06 10-55-18](https://user-images.githubusercontent.com/9427929/177500426-b0b5956b-ef9c-4f5a-98ed-811b88937f58.png)

**Comment by k0machi** [comment]
Please include information from browser console and network tab, that would help to see if it's a temporary issue with the network or something in argus specifically.

**Comment by yarongilor** [comment]
thanks, it is currently not reproduced. i'll update once reproduced.

**Comment by yarongilor** [comment]
reproduced:
```
<div class="vbox flex-auto split-widget" aria-hidden="false" style="--monospace-font-size:11px; --monospace-font-family:dejavu sans mono,monospace; --source-code-font-size:11px; --source-code-font-family:dejavu sans mono,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; position: relative; flex-direction: column !important; flex: 1 1 auto; color: rgb(48, 57, 66); font-family: Roboto, Ubuntu, Arial, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="widget shadow-split-widget hbox" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; flex: 1 1 auto; contain: style; display: flex; flex-direction: row !important; overflow: hidden;"><div class="shadow-split-widget-contents shadow-split-widget-main vbox" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; position: relative; flex-direction: column; contain: size style; flex: 1 1 auto;"><div class="widget vbox" slot="insertion-point-main" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; flex: 1 1 auto; contain: style; display: flex; flex-direction: column !important;"><div class="data-grid no-selection small network-log-grid data-grid-fits-viewport" tabindex="0" role="application" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; line-height: 14.4px; border: none !important; flex: 1 1 auto;"><div class="data-container" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; inset: 0px; overflow: hidden; transform: translateZ(0px); background-color: var(--color-background);">

jobs | 200 | document | Other | 34.0 kB | 423 ms
-- | -- | -- | -- | -- | --
fe29421e-716d-4431-bd2e-a37f3807fd77 | 200 | * | jobs:51 | (memory cache) | 1 ms
main.css | 304 | stylesheet | jobs | 227 B | 120 ms
noto.css | 304 | stylesheet | jobs | 225 B | 153 ms
fontAwesome.css | 304 | stylesheet | jobs | 227 B | 150 ms
fontAwesome.bundle.js | 304 | script | jobs | 228 B | 116 ms
main.bundle.js | 304 | script | jobs | 227 B | 124 ms
globalAlert.bundle.js | 304 | script | jobs | 226 B | 131 ms
argus.png | 304 | png | jobs | 227 B | 178 ms
notificationCounter.bundle.js | 304 | script | jobs | 225 B | 145 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
data:image/svg+xml,… | 200 | svg+xml | index.js:2050 | (memory cache) | 1 ms
profileJobs.bundle.js | 200 | script | jobs | (disk cache) | 27 ms
get_unread | 200 | fetch | ApiUtils.js:5 | 177 B | 263 ms
  |   |   |   |   |  

</div><div class="data-grid-resizer" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; top: 0px; bottom: 0px; width: 5px; z-index: 500; cursor: col-resize; left: 130px;"></div><div class="data-grid-resizer" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; top: 0px; bottom: 0px; width: 5px; z-index: 500; cursor: col-resize; left: 169px;"></div><div class="data-grid-resizer" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; top: 0px; bottom: 0px; width: 5px; z-index: 500; cursor: col-resize; left: 209px;"></div><div class="data-grid-resizer" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; top: 0px; bottom: 0px; width: 5px; z-index: 500; cursor: col-resize; left: 274px;"></div><div class="data-grid-resizer" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; top: 0px; bottom: 0px; width: 5px; z-index: 500; cursor: col-resize; left: 314px;"></div></div></div></div><div class="shadow-split-widget-contents shadow-split-widget-sidebar vbox" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; position: relative; flex-direction: column; contain: size style; flex: 0 0 200px; border-top: 0px; border-right: 0px; border-bottom: 0px; border-image: initial; border-left: 1px solid var(--color-details-hairline);"><slot name="insertion-point-sidebar" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><div class="widget vbox network-waterfall-view" slot="insertion-point-sidebar" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; flex: 1 1 auto; contain: style; display: flex; flex-direction: column !important;"><canvas tabindex="-1" width="184" height="730" style="box-sizing: border-box; min-width: 0px; min-height: 0px; width: 184px; height: 730px;"></canvas><div class="network-waterfall-v-scroll small" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; top: 0px; right: 0px; bottom: 0px; overflow-x: hidden; margin-top: 27px; z-index: 200;"><div class="network-waterfall-v-scroll-content" style="box-sizing: border-box; min-width: 0px; min-height: 0px; width: 15px; pointer-events: none; height: 730px;"></div></div><div class="network-waterfall-header small" style="box-sizing: border-box; min-width: 0px; min-height: 0px; border-bottom: 1px solid var(--color-details-hairline); border-left: 0px; height: 27px; background-color: var(--color-background-elevation-1); position: absolute; width: 199px; display: table; z-index: 200;"><div style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: table-cell; line-height: 14px; margin: auto 0px; vertical-align: middle; text-align: left; font-weight: 400; padding: 0px 4px;">Waterfall</div><div class="sort-order-icon-container" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; line-height: 14px; margin: auto 0px; vertical-align: middle; text-align: left; font-weight: 400; padding: 0px 4px; position: absolute; top: 1px; right: 0px; bottom: 1px; align-items: center;"><span is="ui-icon" class="sort-order-icon spritesheet-smallicons smallicon-triangle-up icon-mask" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: inline-block; flex-shrink: 0; --override-icon-mask-background-color:rgb(110 110 110); background-color: var(--override-icon-mask-background-color); -webkit-mask-position: var(--spritesheet-position); -webkit-mask-image: var(--image-file-smallIcons); align-items: center; margin-right: 4px; margin-bottom: -2px; --spritesheet-position:-20px 10px; width: 10px; height: 10px;"></span></div></div></div></slot></div><div class="shadow-split-widget-resizer" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: absolute; top: 0px; bottom: 0px; width: 6px; z-index: 4000; cursor: ew-resize; touch-action: none; right: 200px; margin-right: -3px;"></div></div></div><div class="network-summary-bar toolbar" role="status" aria-hidden="false" style="flex: 0 0 27px; padding: 0px 2px 0px 5px; --toolbar-height:26px; --monospace-font-size:11px; --monospace-font-family:dejavu sans mono,monospace; --source-code-font-size:11px; --source-code-font-family:dejavu sans mono,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; line-height: 27px; background-color: var(--color-background-elevation-1); border-top: 1px solid var(--color-details-hairline); white-space: nowrap; text-overflow: ellipsis; overflow: hidden; user-select: text; border-bottom: 0px; color: rgb(48, 57, 66); font-family: Roboto, Ubuntu, Arial, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="toolbar-shadow" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; white-space: nowrap; height: var(--toolbar-height); overflow: hidden; z-index: 12; display: flex; flex: 0 0 auto; align-items: center;"><div class="toolbar-text toolbar-item" aria-label="28 requests" title="28 requests" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; display: flex; background-color: transparent; flex: 0 0 auto; align-items: center; justify-content: center; padding: 0px; height: var(--toolbar-height); border: none; white-space: pre; color: var(--color-text-secondary); margin: 0px 5px;">28 requests</div><div class="toolbar-divider toolbar-item" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; display: flex; background-color: var(--color-details-hairline); flex: 0 0 auto; align-items: center; justify-content: center; padding: 0px; height: 16px; border: none; white-space: pre; color: var(--color-text-secondary); width: 1px; margin: 5px 4px;"></div><div class="toolbar-text toolbar-item" aria-label="35970 B transferred over network" title="35970 B transferred over network" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; display: flex; background-color: transparent; flex: 0 0 auto; align-items: center; justify-content: center; padding: 0px; height: var(--toolbar-height); border: none; white-space: pre; color: var(--color-text-secondary); margin: 0px 5px;">36.0 kB transferred</div><div class="toolbar-divider toolbar-item" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; display: flex; background-color: var(--color-details-hairline); flex: 0 0 auto; align-items: center; justify-content: center; padding: 0px; height: 16px; border: none; white-space: pre; color: var(--color-text-secondary); width: 1px; margin: 5px 4px;"></div><div class="toolbar-text toolbar-item" aria-label="3409890 B resources loaded by the page" title="3409890 B resources loaded by the page" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; display: flex; background-color: transparent; flex: 0 0 auto; align-items: center; justify-content: center; padding: 0px; height: var(--toolbar-height); border: none; white-space: pre; color: var(--color-text-secondary); margin: 0px 5px;">3.4 MB resources</div><div class="toolbar-divider toolbar-item" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; display: flex; background-color: var(--color-details-hairline); flex: 0 0 auto; align-items: center; justify-content: center; padding: 0px; height: 16px; border: none; white-space: pre; color: var(--color-text-secondary); width: 1px; margin: 5px 4px;"></div><div class="toolbar-text toolbar-item" aria-label="Finish: 4.17&nbsp;s" title="Finish: 4.17&nbsp;s" style="box-sizing: border-box; min-width: 0px; min-height: 0px; position: relative; display: flex; background-color: transparent; flex: 0 0 auto; align-items: center; justify-content: center; padding: 0px; height: var(--toolbar-height); border: none; white-space: pre; color: var(--color-text-secondary); margin: 0px 5px;">Finish: 4.17 s</div></div></div>
```
```
<ol class="tree-outline hide-selection-when-blurred issues" role="tree" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; padding: 0px; margin: 0px; z-index: 0; position: relative; overflow: auto; --issue-indent:8px; list-style-type: none; color: rgb(48, 57, 66); font-family: Roboto, Ubuntu, Arial, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li role="treeitem" class="issue parent expanded" aria-expanded="true" style="box-sizing: border-box; min-width: 0px; min-height: 16px; padding-top: 0px; padding-right: 8px; padding-bottom: 0px; padding-left: var(--issue-indent); overflow: hidden; flex: 0 0 auto; transition: background-color 0.2s ease 0s; border-top-color: ; border-top-style: ; border-right-color: ; border-right-style: ; border-bottom-color: ; border-bottom-style: ; border-left-color: ; border-left-style: ; border-image-source: ; border-image-slice: ; border-image-width: ; border-image-outset: ; border-image-repeat: ; border-width: 0px; text-overflow: ellipsis; white-space: normal; position: relative; display: flex; align-items: center; background: var(--color-background);"><div class="header" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; flex-direction: row; align-items: center; padding: 6px 0px; cursor: pointer; width: 512.266px;"><div class="title" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 1 1 0%; font-size: 14px; color: var(--color-text-primary); font-weight: 450; user-select: text;">Audit usage of navigator.userAgent, navigator.appVersion, and navigator.platform</div></div></li><ol class="children body issue-kind-improvement expanded" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; list-style-type: none; padding-left: calc(var(--issue-indent) + 43px); border-bottom: 1px solid var(--color-details-hairline-light); padding-top: 6px; position: relative; padding-bottom: 26px; padding-right: 8px; display: block;"><li role="treeitem" class="always-parent" style="box-sizing: border-box; min-width: 0px; min-height: 16px; text-overflow: ellipsis; white-space: normal; position: relative; display: flex; align-items: center;"><span class="tree-element-title" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"></span><devtools-markdown-view style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><div class="message" style="line-height: 20px; font-size: 14px; color: var(--color-text-secondary); margin-bottom: 4px; user-select: text;"><p style="margin-bottom: 16px; margin-block-start: 2px;">A page or script is accessing at least one of<span> </span><code style="color: var(--color-text-primary); font-size: 12px; user-select: text; cursor: text; background: var(--color-background-elevation-1);">navigator.userAgent</code>,<span> </span><code style="color: var(--color-text-primary); font-size: 12px; user-select: text; cursor: text; background: var(--color-background-elevation-1);">navigator.appVersion</code>, and<span> </span><code style="color: var(--color-text-primary); font-size: 12px; user-select: text; cursor: text; background: var(--color-background-elevation-1);">navigator.platform</code>. Starting in Chrome 101, the amount of information available in the User Agent string will be reduced.</p><p style="margin-bottom: 16px; margin-block-start: 2px;">To fix this issue, replace the usage of<span> </span><code style="color: var(--color-text-primary); font-size: 12px; user-select: text; cursor: text; background: var(--color-background-elevation-1);">navigator.userAgent</code>,<span> </span><code style="color: var(--color-text-primary); font-size: 12px; user-select: text; cursor: text; background: var(--color-background-elevation-1);">navigator.appVersion</code>, and<span> </span><code style="color: var(--color-text-primary); font-size: 12px; user-select: text; cursor: text; background: var(--color-background-elevation-1);">navigator.platform</code><span> </span>with feature detection, progressive enhancement, or migrate to<span> </span><code style="color: var(--color-text-primary); font-size: 12px; user-select: text; cursor: text; background: var(--color-background-elevation-1);">navigator.userAgentData</code>.</p><p style="margin-bottom: 16px; margin-block-start: 2px;">Note that for performance reasons, only the first access to one of the properties is shown.</p></div></devtools-markdown-view></li><li role="treeitem" class="always-parent parent expanded affected-resources-label" aria-expanded="true" style="box-sizing: border-box; min-width: 0px; min-height: 16px; margin-top: 5px; font-size: 10px; font-weight: 500; letter-spacing: 0.06em; text-transform: uppercase; color: var(--color-text-primary); display: flex; text-overflow: ellipsis; white-space: normal; position: relative; align-items: center;">AFFECTED RESOURCES</li><ol class="children expanded affected-resources" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; list-style-type: none; padding: 3px 0px 0px; position: relative; user-select: text; display: block;"><li role="treeitem" class="parent expanded" aria-expanded="true" style="box-sizing: border-box; min-width: 0px; min-height: 16px; text-overflow: ellipsis; white-space: normal; position: relative; display: flex; align-items: center; margin-top: 0px; padding: 2px 5px 0px; background: var(--color-background-elevation-0);"><span class="tree-element-title" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"></span><div class="affected-resource-label" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-size: 14px; line-height: 20px; color: var(--color-text-primary); position: relative; cursor: pointer;">1 source</div></li><ol class="children expanded" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; list-style-type: none; padding: 6px 0px 9px 5px; display: block; background: var(--color-background-elevation-0); margin-bottom: 10px;"><li role="treeitem" tabindex="0" class="selected" aria-selected="true" style="box-sizing: border-box; min-width: 0px; min-height: 16px; outline-width: 0px; text-overflow: ellipsis; white-space: normal; position: relative; display: flex; align-items: center;"><span class="tree-element-title" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"></span>

index.js:1
--


</li></ol></ol></ol></ol>
```
![Screenshot from 2022-07-06 16-34-25](https://user-images.githubusercontent.com/9427929/177562633-1a91a227-ab99-4600-9e65-e4bcef80049a.png)
![Screenshot from 2022-07-06 16-32-52](https://user-images.githubusercontent.com/9427929/177562639-ff963651-7d34-4878-befd-ac389c092904.png)

**Comment by k0machi** [comment]
@ShlomiBalalis is also experiencing this, could be related to the amount of requests the browser sends on page load - however, page works normally for him using Firefox - can you check if that also the case for you?

**Comment by k0machi** [comment]
Should be fixed via #191. @yarongilor @ShlomiBalalis can you guys verify that it doesn't freeze for you now?

**Comment by yarongilor** [comment]
> Should be fixed via #191. @yarongilor @ShlomiBalalis can you guys verify that it doesn't freeze for you now?

yes, works good now

**Comment by ShlomiBalalis** [comment]
Works like a charm!
